### PR TITLE
Detect fault ASA configuration

### PR DESCRIPTION
**Pull request by NoamLandress**
Added a function that detects fault Cisco ASA configuration in security_lib and suggests a solution.

### CODE REVIEW

File: DataConnectors/CEF/cef_troubleshoot.py [hunk 1]
```diff
@@ -383,6 +384,23 @@ def check_oms_agent_status():
...
+        Checking if the OMS agent has the updated configuration:
+        :param omsagent_plugin_security_config: path to the file containing the updated configuration
+        :return: True if the configuration is updated, false otherwise
+        '''
+        grep = subprocess.Popen(["grep", "-i", "return ident", oms_agent_plugin_securiy_config], stdout=subprocess.PIPE)
+        o, e = grep.communicate()
+        if not o:
+            print_warning("Warning: Current content of the Omsagent configuration requires update\n")
+            print_notice("Notice: To update the configuration run - \"sed -i \"s|return \'%ASA\'|return ident|g\""
+                         " /opt/microsoft/omsagent/plugin/security_lib.rb\"")
```
Comment (ofshomro) [inline DataConnectors/CEF/cef_troubleshoot.py:398]: you can use oms_agent_plugin_securiy_config

File: DataConnectors/CEF/cef_troubleshoot.py [hunk 2]
```diff
@@ -383,6 +384,23 @@ def check_oms_agent_status():
     else:
         return True
 
+def check_omsagent_new_version(omsagent_plugin_securiy_config):
```
Comment (ofshomro) [inline DataConnectors/CEF/cef_troubleshoot.py:387]: if this is the name maybe we should really check the version of the agent and based our suggestion on that for future updates.

File: DataConnectors/CEF/cef_troubleshoot.py [hunk 3]
```diff
@@ -383,6 +384,23 @@ def check_oms_agent_status():
     else:
         return True
 
+def check_omsagent_new_version(omsagent_plugin_securiy_config):
+        '''
+        Checking if the OMS agent has the updated configuration:
+        :param omsagent_plugin_security_config: path to the file containing the updated configuration
+        :return: True if the configuration is updated, false otherwise
+        '''
+        grep = subprocess.Popen(["grep", "-i", "return ident", oms_agent_plugin_securiy_config], stdout=subprocess.PIPE)
```
Comment (ofshomro) [inline DataConnectors/CEF/cef_troubleshoot.py:393]: Maybe use the full line? "return indent" is something that could be used in the future for other cases.
Also add to the consts

File: DataConnectors/CEF/cef_troubleshoot.py [hunk 4]
```diff
@@ -383,6 +384,23 @@ def check_oms_agent_status():
...
+        '''
+        Checking if the OMS agent has the updated configuration:
+        :param omsagent_plugin_security_config: path to the file containing the updated configuration
+        :return: True if the configuration is updated, false otherwise
+        '''
+        grep = subprocess.Popen(["grep", "-i", "return ident", oms_agent_plugin_securiy_config], stdout=subprocess.PIPE)
+        o, e = grep.communicate()
+        if not o:
+            print_warning("Warning: Current content of the Omsagent configuration requires update\n")
+            print_notice("Notice: To update the configuration run - \"sed -i \"s|return \'%ASA\'|return ident|g\""
```
Comment (ofshomro) [inline DataConnectors/CEF/cef_troubleshoot.py:397]: also full line and consts

**Review by morshabi — APPROVED** [review]

**Pull request merged**